Documentation/ApiOverview/Database/ExpressionBuilder/_RepositoryAs.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Domain\Repository;
+
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Database\Query\QueryBuilder;
+
+final class MyTableRepository
+{
+    private const TABLE_NAME = 'tx_myextension_table';
+    public function __construct(private readonly ConnectionPool $connectionPool) {}
+
+    public function demonstrateExpressionBuilderAs(): QueryBuilder
+    {
+        $queryBuilder = $this->connectionPool
+            ->getQueryBuilderForTable(self::TABLE_NAME);
+        $expressionBuilder = $queryBuilder->expr();
+
+        $queryBuilder->selectLiteral(
```

Comment:
You are demonstrating two variants here, but one is returned, that first looked like unneeded code to me. How about this idea:

```
$variantWithDirectExpression = $queryBuilder->...
$variantWithSubExpression = $queryBuilder->...
return $variantWithSubExpression
```

I am not too happy about picking which querybuilder to return, but due to the return type hint we can't just return void. What do you think?

Reply:
We could also just return void und leave it ad that. Examples are no finished, working code